pkg/pillar/utils/cloudconfig/cloudconfig.go

```diff
@@ -1,14 +1,17 @@
 package cloudconfig
 
 import (
+	"compress/gzip"
```

Comment:
Why not "archive/zip" so that we don't bring new dependencies?

Reply:
It is a matter of user's expectations: from what I've seen so far cloud-init supports the `gzip`, `base64` and `plain` [encodings](https://cloudinit.readthedocs.io/en/latest/reference/examples.html#writing-out-arbitrary-files). I haven't seen the `zip` encoding used in cloud-init.
We can add additional support for `zip`, but the option to decode `gzip` should still be available.